src/utils.jl

```diff
@@ -10,6 +10,20 @@ macro check_args(K, param, cond, desc=string(cond))
 end
 
 
+"""
+    ColVecs{T, TX<:AbstractMatrix}
+
+A lightweight box for an `AbstractMatrix` to make it behave like a vector of vectors.
+"""
+struct ColVecs{T, TX<:AbstractMatrix{T}} <: AbstractVector{SubArray}
+    X::TX
+    ColVecs(X::TX) where {T, TX<:AbstractMatrix{T}} = new{T, TX}(X)
```

Comment:
Can we make the element type concrete, e.g., by using
```suggestion
struct ColVecs{T, TX<:AbstractMatrix{T}, S<:AbstractVector{T}} <: AbstractVector{S}
    X::TX
    function ColVecs(X::TX) where {T, TX<:AbstractMatrix{T}}
        S = typeof(view(X, :, 1))
        return new{T, TX, S}(X)
    end
```
?

Reply:
I am going to try to see if tests are still passing

Reply:
Yay!